src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php

```diff
@@ -247,7 +247,7 @@ public function denormalize($data, $class, $format = null, array $context = arra
 
         $reflectionClass = new \ReflectionClass($class);
         $object = $this->instantiateObject($normalizedData, $class, $context, $reflectionClass, $allowedAttributes, $format);
-
+        $class = get_class($object);
```

Comment:
I'm afraid doing this can cause weird behaviors when using Doctrine proxies, or other kind of proxies.
As a safer trick, I suggest to set the child class name in the context here https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php#L139

Because the context is passed by reference, line 250 we can check if the key exists and replace `$class` value by the child class only in this case. WDYT @sroze?